### PR TITLE
Add securitygroup and loadbalancer unit test

### PR DESCRIPTION
**Pull request by outscale-vbr**
Add securitygroup and loadbalancer unit test:
* Integrate service loadbalancersvc
* Use loadbalancersvc instead of servicesvc
* Add loadbalancer unit test
* Integrate service securitygroupsvc in osccluster_controller
* Use securitygroupsvc instead of securitysvc
* add securityGroup unit test
* Use  setup and setupmock
* Use assert

### CODE REVIEW

**Review by outscale-mdr — COMMENTED** [review]
Good job 👍

File: controllers/osccluster_securitygroup_controller_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,1826 @@
...
+
+// TestCheckSecurityGroupFormatParameters has several tests to cover the code of the function checkSecurityGroupFormatParameters
+func TestCheckSecurityGroupFormatParameters(t *testing.T) {
+	securityGroupTestCases := []struct {
+		name                                     string
+		spec                                     infrastructurev1beta1.OscClusterSpec
+		expCheckSecurityGroupFormatParametersErr error
+	}{
+		{
+			name: "check work without net and securityGroup spec (with default values)",
```
[the remark this answers is not in the document]
Comment (outscale-vbr) [reply controllers/osccluster_securitygroup_controller_unit_test.go]: ok done

File: controllers/osccluster_loadbalancer_controller_unit_test.go [hunk 2]
```diff
@@ -0,0 +1,1023 @@
...
+
+// TestCheckLoadBalancerFormatParameters has several tests to cover the code of the function checkLoadBalancerFormatParameters
+func TestCheckLoadBalancerFormatParameters(t *testing.T) {
+	loadBalancerTestCases := []struct {
+		name                                    string
+		spec                                    infrastructurev1beta1.OscClusterSpec
+		expCheckLoadBalancerFormatParametersErr error
+	}{
+		{
+			name: "check work without loadBalancer spec (with default values)",
```
[the remark this answers is not in the document]
Comment (outscale-vbr) [reply controllers/osccluster_loadbalancer_controller_unit_test.go]: ok done

File: controllers/osccluster_loadbalancer_controller_unit_test.go [hunk 3]
```diff
@@ -0,0 +1,1023 @@
...
+						},
+					},
+					LoadBalancer: infrastructurev1beta1.OscLoadBalancer{
+						LoadBalancerName:  "test-loadbalancer",
+						LoadBalancerType:  "internet-facing",
+						SubnetName:        "test-subnet-test",
+						SecurityGroupName: "test-securitygroup",
+					},
+				},
+			},
```
[the remark this answers is not in the document]
Comment (outscale-vbr) [reply controllers/osccluster_loadbalancer_controller_unit_test.go]: Ok remove

**Review by jerome-jutteau — APPROVED** [review]
LGTM

**Comment by outscale-vbr** [comment]
close https://github.com/outscale-dev/cluster-api-provider-outscale/issues/12

**Pull request merged**